workspaces/libnpmpublish/lib/provenance.js

```diff
@@ -21,7 +21,7 @@ const generateProvenance = async (subject, opts) => {
     /* istanbul ignore next - not covering missing env var case */
     const [workflowPath, workflowRef] = (env.GITHUB_WORKFLOW_REF || '')
       .replace(env.GITHUB_REPOSITORY + '/', '')
-      .split('@')
+      .split('@', 2)
```

Comment:
I'm not sure this will fix the issue.

Using your original string as an example, the original version would give you:

```
'octocat/hello-world/.github/workflows/my-workflow.yml@refs/tags/plugin-vue@1.0.0'.split('@')

[
  'octocat/hello-world/.github/workflows/my-workflow.yml',
  'refs/tags/plugin-vue',
  '1.0.0'
]
```

Using the limit param results in the following:

```
'octocat/hello-world/.github/workflows/my-workflow.yml@refs/tags/plugin-vue@1.0.0'.split('@', 2)

[
  'octocat/hello-world/.github/workflows/my-workflow.yml',
  'refs/tags/plugin-vue'
]
```

In both cases, we've lost the `@1.0.0` bit from the end of the ref.

I think we may need something like this instead:

```
const [workflowPath, ...rest] = (env.GITHUB_WORKFLOW_REF || '')
      .replace(env.GITHUB_REPOSITORY + '/', '')
      .split('@', 2)
const workflowRef = rest.join('@')
```

Reply:
Perhaps a bit more readable . . . 
```
const relativeRef = (env.GITHUB_WORKFLOW_REF || '').replace(env.GITHUB_REPOSITORY + '/', '')
const workflowPath = relativeRef.slice(0, relativeRef.indexOf('@'))
const workflowRef = relativeRef.slice(relativeRef.indexOf('@') + 1)
```